src/plugins/intel_gpu/src/kernel_selector/kernels/scatter_update/scatter_elements_update_kernel_ref.cpp

```diff
@@ -264,34 +281,51 @@ KernelsData ScatterElementsUpdateKernelRef::GetKernelsData(const Params& params)
     GetUpdateDispatchDataFunc(kd);
 
     const auto& output = newParams.outputs[0];
-    const bool use_local_memory = (output.PhysicalSizeInBytes() * 4 > params.engineInfo.maxLocalMemSize) ? false : true;
-    kd.internalBuffers.clear();
-    kd.internalBuffers.push_back(output.PhysicalSizeInBytes() * 2); // fixed point output
-    if (!use_local_memory) {
-        kd.internalBuffers.push_back(output.PhysicalSizeInBytes() * 2); // reduction value output
-        kd.internalBuffers.push_back(output.PhysicalSizeInBytes() * 2); // reduction_thread_count output
-    }
-    kd.internalBufferDataType = Datatype::INT32;
-    if (newParams.mode == ScatterUpdateReduction::MEAN) {
-        kd.internalBuffers.push_back(output.PhysicalSizeInBytes() * 2); // calculate mean
+    const bool use_local_memory_for_static = (output.PhysicalSizeInBytes() * 4 > params.engineInfo.maxLocalMemSize) ? false : true;
```

Comment:
* nit: what about `use_local_memory_for_static_shape`
* could you remove `false : true`? this looks weird